### PR TITLE
Add lazy loading and hydration

### PR DESCRIPTION
**Pull request by filipsobol**
### Short Description and Why It's Useful
Add lazy loading and hydration to improve performance.

### Which Environment This Relates To
- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [X] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [X] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

### CODE REVIEW

File: packages/boilerplate/theme/nuxt.config.js
Comment (filipsobol) [inline packages/boilerplate/theme/nuxt.config.js:29]: Maybe this should be part of `packages/core/nuxt-module/lib/modules/storefront-ui.js` plugin? 🤔

Or maybe fonts shouldn't be part of SFUI, but instead be provided to it by some CSS variable?
@filrak @andrzejewsky @justyna-13 @AdamPawlinski

File: packages/core/nuxt-theme-module/theme/components/CartSidebar.vue
Comment (filipsobol) [inline packages/core/nuxt-theme-module/theme/components/CartSidebar.vue:4]: I think this prop should default to `true`. WDYT @AdamPawlinski and @justyna-13 ?

File: packages/core/nuxt-theme-module/theme/components/LoginModal.vue
Comment (filipsobol) [inline packages/core/nuxt-theme-module/theme/components/LoginModal.vue:3]: I think this prop should default to `true`. WDYT @AdamPawlinski and @justyna-13 ?

File: packages/boilerplate/theme/nuxt.config.js
Comment (filrak) [reply packages/boilerplate/theme/nuxt.config.js:29]: I think it shouldn't because you don't always use these fonts ;)

File: packages/boilerplate/theme/nuxt.config.js
Comment (Fifciu) [inline packages/boilerplate/theme/nuxt.config.js:23]: As you are already using &display=swap, there should be no visual difference with this: https://css-tricks.com/the-fastest-google-fonts/
But Pagespeed should give better results

File: packages/commercetools/theme/nuxt.config.js
Comment (Fifciu) [inline packages/commercetools/theme/nuxt.config.js:24]: Same as above: https://css-tricks.com/the-fastest-google-fonts/

File: packages/core/nuxt-theme-module/theme/components/CartSidebar.vue
Comment (Fifciu) [inline packages/core/nuxt-theme-module/theme/components/CartSidebar.vue:4]: Wouldn't it be cleaner to just have `visible` by default?

File: packages/core/nuxt-theme-module/theme/pages/Home.vue
Comment (Fifciu) [inline packages/core/nuxt-theme-module/theme/pages/Home.vue:18]: Have you checked it manually? I've seen LazyHydrate never on component which uses v-lazy-load and images never loaded

**Review by andrzejewsky — APPROVED** [review]
This is awesome, but please wait until fixes in SFUI will be merged - we can compare then what has changed / improved

File: packages/core/nuxt-theme-module/theme/pages/Product.vue
Comment (Fifciu) [inline packages/core/nuxt-theme-module/theme/pages/Product.vue:182]: Will @click in never hydrated component work?

**Review by Fifciu — CHANGES_REQUESTED** [review]

File: packages/boilerplate/theme/nuxt.config.js
Comment (filrak) [reply packages/boilerplate/theme/nuxt.config.js:23]: if this is preloaded then we don't need a preconnect below

File: packages/core/nuxt-theme-module/theme/components/CartSidebar.vue
Comment (filrak) [reply packages/core/nuxt-theme-module/theme/components/CartSidebar.vue:4]: @AdamPawlinski good idea to have all modals/sidebars visible by default ;)

**Review by filrak — CHANGES_REQUESTED** [review]

File: packages/boilerplate/theme/nuxt.config.js
Comment (filipsobol) [reply packages/boilerplate/theme/nuxt.config.js:22]: No, we are preloading CSS stylesheet, but fonts themselves come from different domain, which we are preconnecting to.

**Review by andrzejewsky — APPROVED** [review]
looks ok, please update changelog

**Review by filrak — APPROVED** [review]

**Pull request merged**
